daft/runners/ray_runner.py

```diff
@@ -684,25 +812,35 @@ def _run_plan(
             f"{datetime.replace(datetime.now(), second=0, microsecond=0).isoformat()[:-3]}.json"
         )
 
-        def is_active():
-            return self.active_by_df.get(result_uuid, False)
-
-        def place_in_queue(item):
-            while is_active():
-                try:
-                    self.results_by_df[result_uuid].put(item, timeout=0.1)
-                    break
-                except Full:
-                    pass
```

Comment:
Got rid of these weird locally-defined callbacks by making them methods, so that I can call them elsewhere without having to pass them around.